test/functional/lua/with_spec.lua

```diff
@@ -0,0 +1,296 @@
+local t = require('test.testutil')
+local n = require('test.functional.testnvim')()
+local Screen = require('test.functional.ui.screen')
+
+local fn = n.fn
+local api = n.api
+local command = n.command
+local eq = t.eq
+local exec_lua = n.exec_lua
+local matches = t.matches
+local pcall_err = t.pcall_err
+
+before_each(function()
+  n.clear()
+end)
+
+describe('vim._with {win = }', function()
```

Comment:
I'd add tests that it doesn't trigger `Win{Enter,Leave}` events. Or at least be explicit and conscious about them (but better the former).